src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs

```diff
@@ -426,7 +426,7 @@ public void Save(Stream stream)
                 WriteManifest(package, DetermineMinimumSchemaVersion(Files, DependencyGroups), psmdcpPath);
 
                 // Write the files to the package
-                HashSet<string> filesWithoutExtensions = new HashSet<string>();
```

Comment:
There's more to the determinism story than just this, see: https://github.com/NuGet/NuGet.Client/blob/99b1006931b1fe11052242be57cd27553019e269/src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs#L34. 

https://github.com/NuGet/Home/issues/8601

I'm surprised this wasn't caught then.

Any chance we can add a test for this?

Reply:
Thanks, I came across https://github.com/NuGet/Home/issues/8601 earlier today (as part of the UID name generation of the psmdcp file), and that seems like a more involved task than this standalone fix. Do you happen to have a way to reproduce the tool failure that lead to turning off deterministic mode? https://github.com/NuGet/Home/issues/8599 is sparse on details.

Reply:
I forget which tool right now, but there were some tools that were using timestamps on the dlls and they didn't push those dlls because they were seen as up to date, ie published is newer than the dlls.

I think some of what deterministic pack was doing is gonna be needed by your work and I those would be safe to make (beyond changing what different versions of the SDK generate). 

> this standalone fix.

Yeah, I agree. I didn't mean to suggest that, mostly saying that you'll probably need more and some of the deterministic flag work might be needed.

Back to other question, can we add a test here?